### PR TITLE
"Varify" more namespaces

### PR DESCRIPTION
**Pull request by bobisageek**
I started at the bottom of that list on babashka/babashka#957 , so I did `clojure.template` and `clojure.lang`. In `clojure.repl`, there are a number of vars that need some extra metadata, so I made `copy-var` have a 3-arg alternative that takes a map of metadata to merge in. That's one reason that I stopped here - I look at that as a significant change, so I'd want that reviewed/renamed before it becomes pervasive. The other reason that I stopped here is that it seems like these changes (in my WSL environment) add about 5MB to the babashka executable size. I'm not sure yet if this is just a product of my environment or what, but again, want to review and take any remediation actions before investing too much time.

Basically, just pausing to have a review conversation. 😄

### CODE REVIEW

**Comment by bobisageek** [comment]
I see node failed - I think what I missed is just requiring that extra macro. I'll get that pushed up in a minute.

**Comment by borkdude** [comment]
Thanks! You don't have to do `clojure.lang` as this is just "fake" stuff to mimic some interfaces from the real Clojure lang.

I'm not a fan of the `merge` approach since it seems all is needed for this is the `:sci.impl/op needs-ctx`, so I would like to see this done like this: https://github.com/borkdude/sci/blob/3c7ebc9adc31261e342145696438022e8a9b9fce/src/sci/impl/namespaces.cljc#L749
Always calling `merge` when it's not actually needed can make this namespace slower to load (both on the JVM and in JS, so I'd like to keep things as fast as possible. (Perhaps my judgement in this case is off, I haven't actually measured the difference).

> add about 5MB to the babashka executable size

This is concerning. Usually adding some docstrings, etc, doesn't add a huge amount of binary size. What are you comparing against, a binary built on the same machine and same GraalVM version? Binary size can vary between GraalVM versions, so that's a thing to note as well. Perhaps you can double check and bisect where this difference comes from.

**Comment by bobisageek** [comment]
There's also one var in repl that needs `{:private true}`, so I guess some options would be 
- taking another arg, `private?` - should be just as fast, but is a bit "smelly" to me, and isn't as extensible
- just leave off the private metadata (which I personally like even less)
- take something like a keyword or a set of keywords which can be checked in the `cond->` block - that might also be a bit slower than specific arguments because of the set lookups, but would also allow for more elegant flexibility over time (if future namespaces need extra/different metadata)

I'm running some builds to bisect the executable size change, so I'll see if these options have any impact on the executable size as well, and maybe I can do some really basic benchmarking to get some ideas about speed as well.

**Comment by borkdude** [comment]
I would say, let's skip the private vars for now. I specifically looked up vars using `ns-publics` in  #957.

**Comment by bobisageek** [comment]
I've made the change form the `merge` call to the optional `assoc`, and that part's good.

However, the executable size is still high, so I'm going to do some experimenting and try to understand (and hopefully eliminate) the root cause of that. I'm with you that just adding some metadata shouldn't have that big of an impact.

**Comment by bobisageek** [comment]
I'm going to re-work this to use your strategy from `core-var`, and rolling back all those `copy-var` changes, so I'm going to close this PR and open a new one from a cleaned-up branch.

A preliminary test of the more limited changes still yields a bigger executable, but I'm getting more and more convinced that the size thing is some WSL anomaly... the sizes have been unpredictable in WSL (but very consistent on my Windows build), and there isn't any gradual growth - there are no 10 or 20 KB size changes - it seems to change either not at all or 5MB.
